docs/operators/kafkaInputOperator.md

```diff
@@ -280,3 +282,145 @@ Below is the configuration for “test” Kafka topic name and
+-   ***repartitionCheckInterval*** - Long
+    -   Interval specified in milliseconds. This value specifies the minimum interval between two stat checks.
+        Default Value = 5 Seconds.
+
+-   ***maxTuplesPerWindow*** - Integer
+    -   Controls the maximum number of messages emitted in each streaming window from this operator. Minimum value is 1. 
+        Default value = MAX_VALUE 
+
+-   ***initialOffset*** - InitialOffset
+    -   Indicates the type of offset i.e, “EARLIEST or LATEST or APPLICATION_OR_EARLIEST or APPLICATION_OR_LATEST”. 
+        LATEST => Operator consume messages from latest point of Kafka queue. 
+        EARLIEST => Operator consume messages starting from message queue.
+        APPLICATION_OR_EARLIEST => Operator consume messages from committed position from last run. If there is no committed offset, then it starts consuming from beginning of kafka queue.
+        APPLICATION_OR_LATEST => Operator consume messages from committed position from last run. If there is not committed offset, then it starts consuming from latest position of queue.
+        Default value = InitialOffset.APPLICATION_OR_LATEST
+
+-   ***metricsRefreshInterval*** - Long
+    -   Interval specified in milliseconds. This value specifies the minimum interval between two metric stat updates.
+        Default value = 5 Seconds.
+
+-   ***consumerTimeout*** - Long
+    -   Indicates the time waiting in poll data if data is not available. Please refer the below link:
+        http://kafka.apache.org/090/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#poll
+        Default value = 5 Seconds.
+
+-   ***holdingBufferSize*** - Long
+    -   Indicates the maximum number of messages kept in memory for emitting.
+        Default value = 1024.
+
+-   ***consumerProps*** - Properties
+    -   Specify the consumer properties which are not yet set to the operator. Please refer the below link for consumer properties:
+        http://kafka.apache.org/090/documentation.html#newconsumerconfigs
+
+-   ***windowDataManager*** - WindowDataManager
+    -   Specifies that the operator will process the same set of messages in a window before and after a failure. This is important but it comes with higher cost because at the end of each window the operator needs to persist some state with respect to that window.
+        Default value = WindowDataManager.NoopWindowDataManager.
+        
+#### Abstract Methods
+
+void emitTuple(String cluster, ConsumerRecord<byte[], byte[]> message): Abstract method that emits tuples
+extracted from Kafka message.
+
+### Concrete Classes
+
+#### KafkaSinglePortInputOperator
+This class extends from AbstractKafkaInputOperator and define the getTuple() method which extracts byte array from Kafka message.
+
+#### Ports
+outputPort <byte[]>: Tuples extracted from Kafka messages are emitted through this port.
+
+### Application Example
+This section builds an Apex application using Kafka input operator.
+Below is the code snippet:
```

Comment:
Instead of putting all the code here, the example will probably be more useful for developers.